pkg/orchestration/wiring/aws/plugin.go

```diff
@@ -38,21 +44,72 @@ func Resource(resourceType voyager.ResourceType,
 	shapes svccatentangler.ShapesFunc,
 ) *WiringPlugin {
 	wiringPlugin := &WiringPlugin{
-		SvcCatEntangler: svccatentangler.SvcCatEntangler{
-			ClusterServiceClassExternalID: clusterServiceClassExternalID,
-			ClusterServicePlanExternalID:  clusterServicePlanExternalID,
-			ResourceType:                  resourceType,
-			Shapes:                        shapes,
-		},
-		OAPResourceTypeName:        oapResourceTypeName,
-		generateServiceEnvironment: generateServiceEnvironment,
+		clusterServiceClassExternalID: clusterServiceClassExternalID,
+		clusterServicePlanExternalID:  clusterServicePlanExternalID,
+		resourceType:                  resourceType,
+		shapes:                        shapes,
+		OAPResourceTypeName:           oapResourceTypeName,
+		generateServiceEnvironment:    generateServiceEnvironment,
 	}
-	wiringPlugin.SvcCatEntangler.InstanceSpec = wiringPlugin.instanceSpec
-	wiringPlugin.SvcCatEntangler.ObjectMeta = wiringPlugin.objectMeta
 	return wiringPlugin
 }
 
-func (awp *WiringPlugin) instanceSpec(resource *orch_v1.StateResource, context *wiringplugin.WiringContext) ([]byte, bool /* externalError */, bool /* retriableError */, error) {
+func (p *WiringPlugin) WireUp(resource *orch_v1.StateResource, context *wiringplugin.WiringContext) wiringplugin.WiringResult {
+	if resource.Type != p.resourceType {
+		return &wiringplugin.WiringResultFailure{
+			Error: errors.Errorf("invalid resource type: %q", resource.Type),
+		}
+	}
+
+	serviceInstance, err := osb.ConstructServiceInstance(resource, p.clusterServiceClassExternalID, p.clusterServicePlanExternalID)
+	if err != nil {
+		return &wiringplugin.WiringResultFailure{
+			Error: err,
+		}
+	}
+
+	instanceParameters, external, retriable, err := p.instanceParameters(resource, context)
+	if err != nil {
+		return &wiringplugin.WiringResultFailure{
+			Error:            err,
+			IsExternalError:  external,
+			IsRetriableError: retriable,
+		}
+	}
+	if instanceParameters != nil {
+		serviceInstance.Spec.Parameters = &runtime.RawExtension{
+			Raw: instanceParameters,
+		}
+	}
+
+	instanceResourceName := wiringutil.ServiceInstanceResourceName(resource.Name)
+
+	smithResource := smith_v1.Resource{
```

Comment:
I find it more readable to merge all the struct literal into a single big one. But it's not a big deal.